src/praisonai/praisonai/agents_generator.py

```
@@ -1247,6 +1247,30 @@ def _run_praisonai(self, config, topic, tools_dict):
+                        # Dict format: {id: "claude-code", overrides: {timeout_ms: 60000}}
+                        backend_id = cli_backend_config.get('id')
+                        overrides = cli_backend_config.get('overrides', {})
+                        if not backend_id:
+                            raise ValueError(f"cli_backend dict must contain 'id' field: {cli_backend_config}")
+                        from praisonai.cli_backends import resolve_cli_backend
+                        cli_backend_resolved = resolve_cli_backend(backend_id, overrides=overrides)
+                    else:
+                        raise ValueError(f"cli_backend must be string or dict, got: {type(cli_backend_config)}")
+                except ImportError:
+                    self.logger.warning(f"CLI backend '{cli_backend_config}' requested but not available")
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)